packages/loot-core/src/server/budget/actions.js

```diff
@@ -4,10 +4,11 @@ import { batchMessages } from '../sync';
 import { groupBySingle } from '../../shared/util';
 import * as prefs from '../prefs';
 import * as sheet from '../sheet';
+import { safeNumber } from "./util";
 
 async function getSheetValue(sheetName, cell) {
   const node = await sheet.getCell(sheetName, cell);
-  return typeof node.value === 'number' ? node.value : 0;
+  return safeNumber(typeof node.value === 'number' ? node.value : 0);
```

Comment:
I don't think we need to do this check when just reading from it; I was thinking a few critical places like this one here: https://github.com/actualbudget/actual/blob/master/packages/loot-core/src/server/budget/rollover.js#L54-L56

I'll chat with you on discord about this!

Reply:
Agreed, we shouldn't have any invalid values here assuming that we've properly checked all writes and arithmetic. It's a relatively cheap check and helpful for debugging in the meantime though.

Reply:
I was thinking that, generically, it's not guaranteed that numbers in the spreadsheet aren't floats. They can be anything, even if realistically we do usually use numbers.

However, this utility _is_ only used by the budget actions so it's cool to leave it in